packages/multichain-network-controller/src/api/accounts-api.test.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  BtcScope,
+  SolScope,
+  EthScope,
+  EthAccountType,
+  BtcAccountType,
+  SolAccountType,
+} from '@metamask/keyring-api';
+import {
+  type Json,
+  type CaipAccountId,
+  KnownCaipNamespace,
+} from '@metamask/utils';
+
+import {
+  type ActiveNetworksResponse,
+  toAllowedCaipAccountIds,
+  toActiveNetworksByAddress,
+  buildActiveNetworksUrl,
+  MULTICHAIN_ACCOUNTS_BASE_URL,
+} from './accounts-api';
+
+const MOCK_ADDRESSES: {
+  evm: string;
+  solana: string;
+  bitcoin: string;
+} = {
+  evm: '0x1234567890123456789012345678901234567890',
+  solana: 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v',
+  bitcoin: 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq',
+};
+
+const MOCK_CAIP_IDS: {
+  evm: CaipAccountId;
+  solana: CaipAccountId;
+  bitcoin: CaipAccountId;
+  invalid: string;
+  invalidEvm: string;
+  invalidSolana: string;
+  invalidBitcoin: string;
+  unsupportedNamespace: string;
+  customNamespace: string;
+  testNamespace: string;
+} = {
+  evm: `${KnownCaipNamespace.Eip155}:1:${MOCK_ADDRESSES.evm}`,
+  solana: `${KnownCaipNamespace.Solana}:1:${MOCK_ADDRESSES.solana}`,
+  bitcoin: `${KnownCaipNamespace.Bip122}:1:${MOCK_ADDRESSES.bitcoin}`,
+  invalid: 'invalid:format:address',
+  invalidEvm: `${KnownCaipNamespace.Eip155}:1:0xinvalid`,
+  invalidSolana: `${KnownCaipNamespace.Solana}:1:invalid`,
+  invalidBitcoin: `${KnownCaipNamespace.Bip122}:1:invalid`,
+  unsupportedNamespace:
+    'cosmos:1:cosmos1hsk6jryyqjfhp5dhc55tc9jtckygx0eph6dd02',
+  customNamespace: `custom:1:${MOCK_ADDRESSES.evm}`,
+  testNamespace: `test:1:${MOCK_ADDRESSES.evm}`,
+};
+
+describe('toAllowedCaipAccountIds', () => {
+  type AccountType = {
+    type: EthAccountType | BtcAccountType | SolAccountType;
+    id: string;
+    options: Record<string, Json>;
+    methods: string[];
+    metadata: {
+      name: string;
+      importTime: number;
+      keyring: { type: string };
+    };
+    address: string;
+    scopes: `${string}:${string}`[];
+  };
+
+  const createMockAccount = (
+    address: string,
+    scopes: `${string}:${string}`[],
+    type: EthAccountType | BtcAccountType | SolAccountType,
+  ): AccountType => ({
```

Comment:
We can re-use `InternalAccount` and `CaipChainId` here.

```suggestion
  const createMockAccount = (
    address: string,
    scopes: CaipChainId[],
    type: InternalAccount['type'],
  ): InternalAccount => ({
```